### PR TITLE
Switch to Txgh:: Namespace

### PR DESCRIPTION
**Pull request by camertron**
This looks like an enormous change, but there are really only a few things going on here:

1. An update to .travis.yml that applies a lumos-specific patch. This patch allows our Travis builds to pass. One of my next tasks will be to adjust the tests so they pass on any fork.
2. All classes and modules are now nested inside the `Txgh::` namespace instead of `Strava::`. Github shows these as deleted and then added, but the files have really just been moved.
3. Removed some `require` statements in favor of autoloading.

@jdoconnor @st33b @zvkemp

### CODE REVIEW

**Comment by camertron** [comment]
P.S. I think this is failing because it's not injecting all the environment variables (probably because this is a pull request and therefore untrusted).

**Comment by camertron** [comment]
Passes here: https://travis-ci.org/lumoslabs/txgh/builds/101692217

**Comment by st33b** [comment]
Looking good

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,9 @@
 language: ruby
 rvm:
-  - 2.1.5
\ No newline at end of file
+  - 2.1.6
+before_install:
+  - |
+    if [ "$LUMOS" == "true" ]; then
```
Comment (jdoconnor) [inline .travis.yml]: this shouldn't be here.

File: lib/txgh.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module Txgh
+  autoload :Application,      'txgh/app'
```
Comment (jdoconnor) [inline lib/txgh.rb]: assuming this is activesupport's autoload, would autoload_at be easier and just pass the path?

**Comment by matthewjackowski** [comment]
@camertron agreed ... I think it's TravisCI as well ... let me run the tests manually first, then I'll merge

File: .travis.yml (hunk 1 above)
Comment (matthewjackowski) [inline .travis.yml]: This is a hard one...currently the tests are setup as essentially 'integration' tests with 1 side of the test mocked and responses returned from Tx or Github to check passing.  SO this approach requires binding  real credentials.  IMHO Instead we should have unit tests in play that can run quickly on PRs and such and a functional test suite to run separately.

Comment (camertron) [inline .travis.yml]: @jdoconnor  I agree. This is a shim until refactoring has reached a point where we don't need separate ENVs to run tests. One of my upcoming goals is to introduce a series of mock objects for API calls.

File: lib/txgh.rb (hunk 2 above)
Comment (camertron) [inline lib/txgh.rb]: I don't believe activesupport is part of the bundle :disappointed:

**Comment by camertron** [comment]
@matthewjackowski are you ok with the `$LUMOS` stuff for now? I **promise** I'll remove it in the near future!

**Comment by matthewjackowski** [comment]
Yep, this is totally fine.

**Comment by matthewjackowski** [comment]
Note: tests passed on master after merge

**Pull request merged**